rosidl_runtime_cpp/include/rosidl_runtime_cpp/bounded_vector.hpp

```diff
@@ -640,6 +640,71 @@ class BoundedVector
   using Base::clear;
 
 private:
+  /// Assign elements from an input range.
+  template<
+    typename InputIterator
+  >
+  void
+  do_assign(InputIterator first, InputIterator last, std::input_iterator_tag)
+  {
+    BoundedVector(first, last).swap(*this);
+  }
+
+  /// Assign elements from a forward range.
+  template<
+    typename FwdIterator
+  >
+  void
+  do_assign(FwdIterator first, FwdIterator last, std::forward_iterator_tag)
+  {
+    if (std::distance(first, last) > UpperBound) {
+      throw std::length_error("Exceeded upper bound");
+    }
+    Base::assign(first, last);
+  }
+
+  // Insert each value at the end and then rotate them to the desired position.
+  // If the bound is exceeded, the inserted elements are removed again.
+  template<
+    typename InputIterator
+  >
+  typename Base::iterator
+  do_insert(
+    typename Base::const_iterator position,
+    InputIterator first,
+    InputIterator last,
+    std::input_iterator_tag)
+  {
+    const auto orig_size = size();
+    const auto idx = position - cbegin();
+    try {
+      while (first != last)
+        push_back(*first++);
```

Comment:
@jwakely nit: use braces for this `while` statement's block

Reply:
Fixed and force-pushed.